Automattic/wp-calypso
# Hides the Promote Post header in the WC Mobile App

We want to hide the Promote post header when the page is loaded from the WooCommerce app. We are already doing it for the jetpack mobile app.

![Screenshot 2023-12-12 at 15 54 47](https://github.com/Automattic/wp-calypso/assets/1258162/f9551ec7-825c-4a67-9b35-801edf52aa27)


## Proposed Changes

* Add another flag in the server to validate the source against WooCommerce user agent
* When the document is loaded, add the class `is-mobile-app-view` to the body if its Jetpack or WooCommerce mobile app

## Testing Instructions

* Open the live preview and navigate to Tools->Advertising
* In your browser, change the page to a mobile device, and change the user agent to be this one:
```
Mozilla/5.0 (iPhone; CPU iPhone OS 14_6 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Mobile/15E148 wc-android
```
* Refresh the page and verify that the Woo logo is shown on the page load
![Screenshot 2023-12-15 at 16 27 11](https://github.com/Automattic/wp-calypso/assets/1258162/223868a4-306e-4bb0-8823-ea2aff4c66c2)

* Verify that the Advertising dashboard is displayed correctly for a mobile app (no header and menu)
* Click on the Promote post for any of your post and verify that the Widget header is hidden and that the Get started step was not shown


## Pre-merge Checklist
